qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java

```diff
@@ -0,0 +1,224 @@
+    }
+
+    /**
+     * Send some basic index, count and delete requests, in order to check that the installation
+     * is minimally functional.
+     */
+    public void test50BasicApiTests() throws Exception {
+        waitForElasticsearch();
+
+        assertTrue(existsInContainer(installation.logs.resolve("gc.log")));
+
+        ServerUtils.runElasticsearchTests();
+    }
+
+    /**
+     * Check that the default keystore is automatically created
+     */
+    public void test60AutoCreateKeystore() throws Exception {
+        final Path keystorePath = installation.config("elasticsearch.keystore");
+
+        waitForPathToExist(keystorePath);
+
+        assertPermissionsAndOwnership(keystorePath, p660);
+
+        final Installation.Executables bin = installation.executables();
+        final Result result = sh.run(bin.elasticsearchKeystore + " list");
+        assertThat(result.stdout, containsString("keystore.seed"));
+    }
+
+    /**
+     * Check that the default config can be overridden using a bind mount, and that env vars are respected
+     */
+    public void test70BindMountCustomPathConfAndJvmOptions() throws Exception {
+        final Path tempConf = getTempDir().resolve("esconf-alternate");
+
+        try {
+            mkdir(tempConf);
+            copyFromContainer(installation.config("elasticsearch.yml"), tempConf.resolve("elasticsearch.yml"));
+            copyFromContainer(installation.config("log4j2.properties"), tempConf.resolve("log4j2.properties"));
+
+            // we have to disable Log4j from using JMX lest it will hit a security
+            // manager exception before we have configured logging; this will fail
+            // startup since we detect usages of logging before it is configured
+            final String jvmOptions =
+                "-Xms512m\n" +
+                "-Xmx512m\n" +
+                "-Dlog4j2.disable.jmx=true\n";
+            append(tempConf.resolve("jvm.options"), jvmOptions);
+
+            // Make the temp directory and contents accessible when bind-mounted
+            Files.setPosixFilePermissions(tempConf, fromString("rwxrwxrwx"));
+
+            // Restart the container
+            removeContainer();
+            runContainer(distribution(), tempConf, Map.of(
+                "ES_JAVA_OPTS", "-XX:-UseCompressedOops"
+            ));
+
+            waitForElasticsearch();
+
+            final String nodesResponse = makeRequest(Request.Get("http://localhost:9200/_nodes"));
+            assertThat(nodesResponse, containsString("\"heap_init_in_bytes\":536870912"));
+            assertThat(nodesResponse, containsString("\"using_compressed_ordinary_object_pointers\":\"false\""));
+        } finally {
+            rm(tempConf);
+        }
+    }
+
+    /**
+     * Check whether the elasticsearch-certutil tool has been shipped correctly,
+     * and if present then it can execute.
+     */
+    public void test90SecurityCliPackaging() {
```

Comment:
We definitely don't want to have copies of every test in archive tests. Most of that class needs to be broken up. I wonder if in the interum ArchiveTests could be adapted to run when docker type is used?